server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/model/MailboxFactory.java

```diff
@@ -168,7 +168,8 @@ private Mailbox from(MailboxId mailboxId,
                          QuotaLoader quotaLoader,
                          MailboxSession mailboxSession) throws MailboxException {
         boolean isOwner = mailboxPath.belongsTo(mailboxSession);
-        Optional<Role> role = Role.from(mailboxPath.getName());
+        Optional<Role> role = Role.from(mailboxPath.getName())
+            .filter(any -> mailboxPath.belongsTo(mailboxSession));
```

Comment:
From @mbaechler in #3392 :

> I can't see any change in MailboxFactoryTest for this change here

> You lack a test in MailboxFactoryTest. You wrote a test at a higher level but it's not enough in my opinion

Reply:
I don't see any test related to shared mailboxes in the lower level `MailboxFactoryTest`. My guess is because it's probably complicated to modify the ACL rights of a mailbox at that level? (might be wrong though). So why should I add this test in the lower `MailboxFactoryTest`?

Reply:
> So why should I add this test in the lower MailboxFactoryTest?

Because you can't really rely on consumer to test your components.
The last thing I want is to have very slow tests far from my code to fail because it's unit test suite is not good enough.

Remember the test pyrami